engine/src/main/java/com/datatorrent/stram/client/StramClientUtils.java

```diff
@@ -519,6 +520,52 @@ public static FileSystem newFileSystemInstance(Configuration conf) throws IOExce
     }
   }
 
+  /**
+   * This gets the DFS Root dir to be used by Apex applications as per the following logic:
+   *    - if Apex-root-dir is null then just call getDTDFSRootDir to get the old behavior
+   *    - if apex.dfsRootDirectory defined use it, otherwise fall back to using dt.dfsRootDirectory by calling getDTDFSRootDir.
+   *    - Value of apex.dfsRootDirectory is referred to as Apex-root-dir below.
+   *    - if Apex-root-dir value is an absolute path then scan for %USER_NAME% in the string and replace it with the user's name, else use the absolute path as is.
+   *    - if Apex-root-dir value is a relative path then append it to the user's home directory.
+   *
+   * In the above steps a "user" refers to either impersonating or impersonated user.
+   * If the value of apex.application.path.impersonated is true we will use the impersonated user else impersonating user.
+   *
+   * @param fs
+   * @param conf
+   * @param impersonated user
+   *
+   * @return
+   */
+  public static Path getApexDFSRootDir(FileSystem fs, Configuration conf, String impersonated)
+  {
+    String apexDfsRootDir = conf.get(APEX_DFS_ROOT_DIR);
+    if (StringUtils.isBlank(apexDfsRootDir)) {
+      return getDTDFSRootDir(fs, conf);
+    }
+    try {
+      boolean useImpersonated = conf.getBoolean(StramUserLogin.DT_APP_PATH_IMPERSONATED, false);
+      String userShortName = useImpersonated ? impersonated : UserGroupInformation.getLoginUser().getShortUserName();
+      if (apexDfsRootDir.startsWith("/")) {
+        if (apexDfsRootDir.contains(DT_DFS_USER_NAME)) {
+          apexDfsRootDir = apexDfsRootDir.replace(DT_DFS_USER_NAME, userShortName);
+          conf.set(DT_DFS_ROOT_DIR, apexDfsRootDir);
+        }
+      } else {
+        URI uri = new URI(apexDfsRootDir);
+        if (uri.isAbsolute()) {
+          return new Path(uri);
+        }
+        apexDfsRootDir = "/user/" + userShortName + "/" + apexDfsRootDir;
```

Comment:
Is it needed to convert it to absolute path. getDTDFSRootDir doesn't seem to be doing that.